pkg/notes/internal/retry.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"time"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	MaxGithubRetries = 3
+
+	defaultGithubSleep = time.Minute
+)
+
+func GithubRetryer(maxTries int, sleeper func(time.Duration)) func(error) bool {
+	try := 0
+
+	return func(err error) bool {
+		if err == nil {
+			return false
+		}
+		if try >= maxTries {
+			logrus.
+				WithField("err", err).
+				Errorf("Exhausted %d tries, not retrying anymore", try)
```

Comment:
I think we should stick to something like:
```suggestion
				Errorf("Exhausted %d tries, not retrying anymore: %v", try, err)
```
instead of using `WithField`, which are mostly consumed by humans in our case.